addons/binding/org.openhab.binding.bosesoundtouch10/src/main/java/org/openhab/binding/bosesoundtouch/handler/BoseSoundTouchHandler.java

```diff
@@ -0,0 +1,397 @@
+                    switch (xmlResult.getValue()) {
+                        case SOURCE_STANDBY:
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_POWER, OnOffType.OFF);
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_NOW_PLAYING, new StringType(""));
+                            break;
+                        case SOURCE_BLUETOOTH:
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_POWER, OnOffType.ON);
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_CONTROL, PlayPauseType.PLAY);
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_SOURCE,
+                                    new StringType(SOURCE_BLUETOOTH));
+                            break;
+                        case SOURCE_AUX:
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_POWER, OnOffType.ON);
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_CONTROL, PlayPauseType.PLAY);
+                            updateState(BoseSoundTouchBindingConstants.CHANNEL_SOURCE, new StringType(SOURCE_AUX));
+                            break;
+                        default:
+                            break;
+                    }
+                    break;
+                case XmlHelper.WS_UPDATE_BASS:
+                    getBassAndUpdateUI();
+                    break;
+                case XmlHelper.WS_UPDATE_SELECTION:
+                    String id = xmlResult.getValue();
+                    if (!id.isEmpty()) {
+                        updateState(BoseSoundTouchBindingConstants.CHANNEL_SOURCE,
+                                new StringType(SOURCE_PRESET_PREFIX + id));
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public void setStatusOffline() {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Websocket error");
+        socketListener = null;
+        try {
+            if (client != null) {
+                client.stop();
+            }
+            client = null;
+        } catch (Exception e) {
+            logger.debug("Error on stopping websocketclient");
+        }
+    }
+
+    private boolean createAndStartWebsocket() {
+        if (soundTouchURL != null && !soundTouchURL.isEmpty() && !soundTouchURL.startsWith("0.0.0.0")) {
+            client = new WebSocketClient();
+            try {
+                client.start();
+                URI echoUri = new URI(soundTouchURL.replace(PROTOCOL_HTTP, PROTOCOL_WS).replace(HTTP_PORT, WS_PORT));
+                ClientUpgradeRequest request = new ClientUpgradeRequest();
+                request.setSubProtocols("gabbo");
+                client.connect(socketListener, echoUri, request);
+                return true;
+            } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception